odoo/addons/test_testing_utilities/tests/test_form_impl.py

```diff
@@ -108,6 +109,108 @@ def test_attrs(self):
+                    </form>
+                """,
+            })
+
+        with self.assertRaises(ValidationError):
+            # a: base.group_no_one > -
+            # d: base.group_no_one > base.group_erp_manager
+            self.env['ir.ui.view'].create({
+                'name': 'stuff',
+                'model': 'test_testing_utilities.groups',
+                'arch': """
+                    <form>
+                        <field name="a" readonly="d"/>
+                    </form>
+                """,
+            })
+
+        with self.assertRaises(ValidationError):
+            # e: base.group_no_one > base.group_erp_manager,base.group_portal
+            # d: base.group_no_one > base.group_erp_manager
+            self.env['ir.ui.view'].create({
+                'name': 'stuff',
+                'model': 'test_testing_utilities.groups',
+                'arch': """
+                    <form>
+                        <field name="e" readonly="d"/>
+                    </form>
+                """,
+            })
+
+        with self.assertRaises(ValidationError):
+            # i: base.group_no_one > !base.group_portal
+            # h: base.group_no_one > base.group_erp_manager,!base.group_portal
+            self.env['ir.ui.view'].create({
+                'name': 'stuff',
+                'model': 'test_testing_utilities.groups',
+                'arch': """
+                    <form>
+                        <field name="i" readonly="h"/>
+                    </form>
+                """,
+            })
+
+        with self.assertRaises(ValidationError):
+            # i: base.group_no_one > !base.group_portal
+            # h: base.group_no_one > base.group_portal
+            self.env['ir.ui.view'].create({
+                'name': 'stuff',
+                'model': 'test_testing_utilities.groups',
+                'arch': """
+                    <form>
+                        <field name="i" readonly="j"/>
+                    </form>
+                """,
+            })
+
+    def test_related_field_and_groups(self):
+        with self.assertRaises(ValidationError):
+            # group from related
+            self.env['ir.ui.view'].create({
+                'name': 'stuff',
+                'model': 'test_testing_utilities.groups_sub',
+                'arch': """
+                    <form>
+                        <field name="g_id"/>
+                    </form>
+                """,
+            })
+
+        with self.assertRaises(ValidationError):
+            # fail because can not add 'g_d' used from related domain
+            self.env['ir.ui.view'].create({
+                'name': 'stuff',
+                'model': 'test_testing_utilities.groups_sub_sub',
+                'arch': """
+                    <form>
+                        <field name="g_sub_id" groups="base.group_erp_manager"/>
+                    </form>
+                """,
+            })
```

Comment:
I think the validation error is not the one you think it is, `g_sub_id` doesn't exist.

That's why I like to always assert the content of the error (like with `assertRaisesRegex`)